core/src/xml/tree.rs

```diff
@@ -14,80 +14,95 @@ use std::collections::BTreeMap;
 use std::fmt;
 use std::mem::swap;
 
+enum XmlNodeType {
```

Comment:
I wonder if we should keep these as `const`s instead of an `enum`, because it can be any arbitrary value and avoids `as u8` in most places.